src/mpb.cpp

```diff
@@ -591,6 +600,28 @@ void *fields::get_eigenmode(double frequency, direction d, const volume where, c
     } while (match_frequency &&
              fabs(sqrt(eigvals[band_num - 1]) - frequency) > frequency * match_tol);
 
+  if (am_master() && dp) {
+    scalar_complex s = {real(dp->get_s()),imag(dp->get_s())};
+    scalar_complex p = {real(dp->get_p()),imag(dp->get_p())};
+
+    master_printf("diffracted planewave: k = %g, %g, %g for omega = %g\n", k[0],k[1],k[2], frequency);
```

Comment:
It needs to first compute the correct k vector from the frequency and `get_g()`.